src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9276,6 +9287,114 @@ def quantiles_along_axis0(
                 ],
             )
         )
+        return result
+
+    def _quantiles_single_col(
+        self,
+        q: list[float],
+        interpolation: Literal["linear", "lower", "higher", "midpoint", "nearest"],
+        index: Optional[Union[list[str], list[float]]] = None,
+        index_dtype: Optional[npt.DTypeLike] = None,
+    ) -> "SnowflakeQueryCompiler":
+        """
+        Helper method for qcut() + quantile() to compute quantiles over frames with a single column.
+
+        Normally, we compute single row for every given quantile, with each column corresponding to
+        a column to the input frame.
```

Comment:
really love the description here, great job!

Reply:
+100